### ISSUE TITLE
array-element-newline direction

### ISSUE THREAD

**Issue by Crusader4Christ** [eslint/eslint#13491]
**What rule do you want to change?** 
array-element-newline
**Does this change cause the rule to produce more or fewer warnings?**
fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
New option 
```
direction: string = fold | unfold | both (default value)
```
**Please provide some example code that this change will affect:**
We need some rule to split long arrays by one element on line, but we have some very long items in small arrays, so we can't use "minItems' option, becuase this rule combine this small arrays with big elements. 

```js
var arrWithLongStrings = [
  'one item is very long string, so it should be on separate line',
  'another item is very long string, so it should be on separate line',
];
var simpleSmallArray = [
1,
2,
];
var simpleLongArray = [1,2,3,4];
```
So we need 
```
/*eslint array-element-newline: ["error", { "multiline": true, { "minItems": 3 }, "direction":"unfold" }]*/
```
This way rule will split arrays, but never combine.

**What does the rule currently do for this code?**
```
var arrWithLongStrings = ['one item is very long string, so it should be on separate line',  'another item is very long string, so it should be on separate line'];
var simpleSmallArray = [1, 2];
var simpleLongArray = [
  1,
  2,
  3,
  4,
];
```
**What will the rule do after it's changed?**
```
var arrWithLongStrings = [
  'one item is very long string, so it should be on separate line',
  'another item is very long string, so it should be on separate line',
];
var simpleSmallArray = [
  1,
  2
];
var simpleLongArray = [
  1,
  2,
  3,
  4,
];
```

**Comment by anikethsaha** [comment]
Thanks for the issue 👍 

`array-element-newline` is a [stylistic](https://github.com/eslint/eslint/blob/master/lib/rules/array-element-newline.js#L20) rule and as per our [new policy](https://eslint.org/blog/2020/05/changes-to-rules-policies#what-are-the-changes) no new enhancement for stylistic rules are being accepted. 

You can create a custom rule from this rule using [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer#readme) or extend this rule as a [base rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/brace-style.ts#L2)

**Comment by Crusader4Christ** [comment]
https://github.com/eslint/eslint/pull/13492
@anikethsaha 
Pity. I've already done PR (without any new testcases but it is working)

**Comment by mdjermanovic** [comment]
Closing this issue, as explained in https://github.com/eslint/eslint/pull/13492#issuecomment-667411579
